src/components/input-time-picker/input-time-picker.tsx

```diff
@@ -84,12 +92,39 @@ export class InputTimePicker
 
     if (value) {
       this.reposition(true);
+
+      activateFocusTrap(this, {
+        onActivate: () => {
+          if (this.focusOnOpen) {
+            this.calciteTimePickerEl.setFocus();
+            this.focusOnOpen = false;
+          }
+        }
+      });
+    } else {
+      deactivateFocusTrap(this);
```

Comment:
Similar to `activateFocusTrap` above, can you pass an options object with `onDeactivate` where you set the following lines (restore focus and set `focusOnOpen`)? Just noticed we should do this for `input-date-picker` as well (separately).